File: backend/lib/collections.py
"""Five Qdrant collections (Chapter 9 naming)."""

VECTOR_SIZE = 1024
DENSE_NAME = "dense"
SPARSE_NAME = "sparse"

SELENIUM_CODE = "selenium_code"
PLAYWRIGHT_CODE = "playwright_code"
VWO_TESTCASES = "vwo_testcases"
VWO_DOCS = "vwo_docs"
VWO_BUGS = "vwo_bugs"

ALL_COLLECTIONS = (
    SELENIUM_CODE,
    PLAYWRIGHT_CODE,
    VWO_TESTCASES,
    VWO_DOCS,
    VWO_BUGS,
)

SOURCE_LABELS = {
    SELENIUM_CODE: "Selenium TestNG Java",
    PLAYWRIGHT_CODE: "Playwright TS/JS",
    VWO_TESTCASES: "VWO test cases (CSV)",
    VWO_DOCS: "VWO PRDs / PDFs",
    VWO_BUGS: "VWO JIRA exports (Markdown)",
}

File: frontend/src/lib/collections.ts
export const COLLECTION_PILLS: { id: string; label: string }[] = [
  { id: "selenium_code", label: "Selenium" },
  { id: "playwright_code", label: "Playwright" },
  { id: "vwo_testcases", label: "Test Cases" },
  { id: "vwo_docs", label: "PRDs / Docs" },
  { id: "vwo_bugs", label: "JIRA Bugs" },
];

export function routerCaption(collections: string[]): string {
  if (collections.length === 0) return "No collections selected.";
  if (collections.length > 2) return "Searching all knowledge bases.";
  const c = collections[0];
  if (c === "vwo_testcases") return "Filtering test cases by priority and module";
  if (c === "vwo_docs") return "Searching product requirements and PDF documentation";
  if (c === "vwo_bugs") return "Searching JIRA bug exports and markdown notes";
  if (c === "selenium_code") return "Searching Selenium Java / TestNG automation code";
  if (c === "playwright_code") return "Searching Playwright TypeScript test framework";
  return `Searching ${collections.join(" and ")}`;
}

export function chunkDisplayId(pointId: unknown, index: number): string {
  if (pointId != null && String(pointId) !== "") return `r${pointId}-c0`;
  return `r${index}-c0`;
}
